packages/react/src/components/CardLink/CardLink.js

```diff
@@ -36,13 +44,34 @@ const CardLink = ({ title, href, content, icon, className, ...props }) => {
       className={classNames(`${prefix}--card-link`, className)}
       href={href}
       {...props}>
+      {renderImage(source)}
       <h3 className={`${prefix}--card-link__title`}>{title}</h3>
       {optionalContent(content)}
       {renderFooter(icon)}
     </ClickableTile>
   );
 };
 
+/**
+ * Render image
+ *
+ * @param {string} source passes as src
+ * @returns {object} JSX object
+ */
+function renderImage(source) {
+  if (!source) {
+    return null;
+  }
+
+  return (
+    <img
+      src={source}
```

Comment:
`source` seems a bit too vague. Should use something like `imgSrc` and `imgAltText`, but check with @photodow

Reply:
Sure @jeffchew

Reply:
Any reason why we can't just do 
```
{source && <img src={source} classname={`${prefix}--card-link__image`} alt={alt} />}
```

or is it cleaner to look at with the `renderImage` func?

Reply:
@jeffchew I've been wondering if there is a way we could unify the data model to be a little more consistent across the board. Almost everything (up to this point) has a very similar content structure, just some things are optional. For an image (still in progress #767) I think it would be great if we could use something like this.

```js
{
  src: "https://example.com/img.jpg",
  alt: "example image"
  ...
  // anything else needed for the image (e.g. maybe aria attributes, or title?)
  ...
}
```

So it would be a little more like this... `<img src="image.src" alt="image.alt"/>`. Or you could even use a spread here I would think, maybe... `<img {...image} />`

Reply:
@photodow is the idea to use the responsive image component for these types of things, just have a bunch of optional fields for it?

Reply:
Yes, that's what I'm hoping for, so we can keep things consistent.